cli/src/commands/generate/components/node.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+
+import { initLogger } from '../../helper.js';
+import { SchemaDirectory } from '../schema-directory.js';
+import { mergeSchemas } from '../util.js';
+import { getPropertyValue } from './property.js';
+
+/**
+ * Instantiate an individual node from its definition, resolving $refs if appropriate. 
+ * @param nodeDef The definition of the node.
+ * @param schemaDirectory The schema directory to resolve refs against.
+ * @param debug Whether to log debug detail.
+ * @returns An instantiated node.
+ */
+export function instantiateNode(nodeDef: any, schemaDirectory: SchemaDirectory, debug: boolean = false): any {
```

Comment:
I also wonder if we can do better on the `nodeDef` type... for example we know that it might have a `$ref` key, so we could at least have `{ '$ref'?: string }` and whatever other optional fields there may be